### PR TITLE
[4.x] Add isSuccess to Http/Client/Response

### PR DESCRIPTION
**Pull request by Iandenh**
refs: #11990 #11536 #11796

### CODE REVIEW

File: src/Http/Client/Response.php [hunk 1]
```diff
@@ -236,6 +236,16 @@ public function isOk()
         return $this->code >= 200 && $this->code <= 399;
     }
 
+    /**
+     * Check if the response status code was in the 2xx range
+     *
+     * @return bool
+     */
+    public function isSuccessful()
```
Comment (ADmad) [inline src/Http/Client/Response.php]: Nitpick: Can we have a shorter name `isSuccess()`? :slightly_smiling_face:

Comment (Iandenh) [reply src/Http/Client/Response.php]: Sure

File: tests/TestCase/Http/Client/ResponseTest.php [hunk 2]
```diff
@@ -244,6 +244,77 @@ public function testIsOk()
...
+                    'HTTP/1.1 203 Non-Authoritative Information',
+                    'Content-Type: text/html'
+                ], 'ok')
+            ],
+            [
+                true,
+                new Response([
+                    'HTTP/1.1 204 No Content',
+                    'Content-Type: text/html'
+                ], 'ok')
```
Comment (markstory) [inline tests/TestCase/Http/Client/ResponseTest.php]: Why does 204 have content?

Comment (Iandenh) [reply tests/TestCase/Http/Client/ResponseTest.php]: Sorry, just copy the test data from `testIsOk`. Shall I also remove the content from the response there?

**Pull request merged**